src/software/sensor_fusion/sensor_fusion_test.cpp

```diff
@@ -194,35 +221,89 @@ TEST_F(SensorFusionTest, test_complete_wrapper_with_tbots_robot_msg_1_at_a_time)
     SensorMsg sensor_msg_1;
     auto ssl_wrapper_packet =
         createWrapperPacket(std::move(geom_data), std::move(detection_frame));
-    *(sensor_msg_1.mutable_ssl_vision_msg()) = *(ssl_wrapper_packet.release());
-    *(sensor_msg_1.add_tbots_robot_msgs())   = *(tbots_robot_msg_id_1.release());
+    *(sensor_msg_1.mutable_ssl_vision_msg()) = *ssl_wrapper_packet;
+    *(sensor_msg_1.add_tbots_robot_msgs())   = *tbots_robot_msg_id_1;
     sensor_fusion.updateWorld(sensor_msg_1);
     EXPECT_NE(std::nullopt, sensor_fusion.getWorld());
     ASSERT_TRUE(sensor_fusion.getWorld());
     World result_1 = *sensor_fusion.getWorld();
-    // TODO (issue #1149): Add checks on the state of World
+    // TODO (Issue #1276): Add checks on the state of World
     SensorMsg sensor_msg_2;
-    *(sensor_msg_2.add_tbots_robot_msgs()) = *(tbots_robot_msg_id_2.release());
+    *(sensor_msg_2.add_tbots_robot_msgs()) = *tbots_robot_msg_id_2;
     sensor_fusion.updateWorld(sensor_msg_2);
     World result_2 = *sensor_fusion.getWorld();
-    // TODO (issue #1149): Add checks on the state of World
+    // TODO (Issue #1276): Add checks on the state of World
 }
 
 TEST_F(SensorFusionTest, test_complete_wrapper_with_tbots_robot_msg_2_at_a_time)
 {
     SensorMsg sensor_msg_1;
     auto ssl_wrapper_packet =
         createWrapperPacket(std::move(geom_data), std::move(detection_frame));
-    *(sensor_msg_1.mutable_ssl_vision_msg()) = *(ssl_wrapper_packet.release());
+    *(sensor_msg_1.mutable_ssl_vision_msg()) = *ssl_wrapper_packet;
     sensor_fusion.updateWorld(sensor_msg_1);
     EXPECT_NE(std::nullopt, sensor_fusion.getWorld());
     ASSERT_TRUE(sensor_fusion.getWorld());
     World result_1 = *sensor_fusion.getWorld();
-    // TODO (issue #1149): Add checks on the state of World
+    // TODO (Issue #1276): Add checks on the state of World
+    SensorMsg sensor_msg_2;
+    *(sensor_msg_2.add_tbots_robot_msgs()) = *tbots_robot_msg_id_1;
+    *(sensor_msg_2.add_tbots_robot_msgs()) = *tbots_robot_msg_id_2;
+    sensor_fusion.updateWorld(sensor_msg_2);
+    World result_2 = *sensor_fusion.getWorld();
+    // TODO (Issue #1276): Add checks on the state of World
+}
+
+TEST_F(SensorFusionTest, test_referee_yellow_then_normal)
+{
+    // TODO (Issue #960): Dependency inject config so that we don't have to implicitly
+    // assume that friendly is yellow
+    GameState expected_1;
+    expected_1.updateRefboxGameState(RefboxGameState::INDIRECT_FREE_US);
+
+    GameState expected_2 = expected_1;
+    expected_2.updateRefboxGameState(RefboxGameState::NORMAL_START);
+
+    SensorMsg sensor_msg_1;
+    auto ssl_wrapper_packet =
+        createWrapperPacket(std::move(geom_data), std::move(detection_frame));
+    // set vision msg so that world is valid
+    *(sensor_msg_1.mutable_ssl_vision_msg()) = *ssl_wrapper_packet;
+    *(sensor_msg_1.mutable_ssl_refbox_msg()) = *referee_indirect_yellow;
```

Comment:
Can we use the "convention" we have been following here (and in the rest of the file)?

`sensor_msg_1.set_allocated_ssl_vision_msg(ssl_wrapper_packet.release())`